### ISSUE TITLE
"String 'path' does not match pattern" error for a path returned by Dropbox itself

### ISSUE THREAD

**Issue by jacobras** [dropbox/dropbox-sdk-java#52]
In my app I'm doing this:

```java
String path = entry.getPathLower();
DbxDownloader<FileMetadata> downloader = mDbxClient.files().download(path);
```

Where `entry` is a `com.dropbox.core.v2.filesFileMetadata` instance returned by the SDK using `listFolderBuilder()`. In Crashlytics I'm seeing `IllegalArgumentException` errors:

`IllegalArgumentException: String 'path' does not match pattern'.`

Which are being thrown from the constructor of `com.dropbox.core.v2.filesDownloadArg`:

```java
if (!java.util.regex.Pattern.matches("((/|id:).*)|(rev:[0-9a-f]{9,})", path)) {
    throw new IllegalArgumentException("String 'path' does not match pattern");
}
```

**The issue:** Dropbox appears to be returning metadata objects with an invalid path. It doesn't occur that often but it's neither rare. Also, users get this error multiple times, so it doesn't seem to happen sporadically.

In the logs I'm receiving I don't see any special characters or something like that, but it could be that Crashlytics strips them out. I've had some issues with uploading where users managed to get `\n` characters inside the filename, perhaps the SDK has a similar issue?

**Comment by greg-db** [comment]
Thanks for the report! It looks like some users may have files in their accounts with paths that don't match the API validation pattern (/.*), such as with new lines. We'll look into it.

**Comment by jacobras** [comment]
Thanks @greg-db! If you find something or if there's something I can do in my app to work around it, please let me know. For new uploads I'm already stripping out newlines but since this is the path returned from Dropbox I don't know what kind of invalid characters are in there.

**Comment by greg-db** [comment]
Unfortunately I don't think there's a workaround that would enable you to download the file, but you can improve the failure mode by catching `IllegalArgumentException` at least.

**Comment by jacobras** [comment]
I'm already doing that, but thanks for the suggestion. I'll keep an eye out for updates on this issue.

**Comment by jacobras** [comment]
Any update on this yet @greg-db? I received 340 reports of this issue (through Crashlytics) in the last 30 days. All of them are the same: trying to download a path that was returned by the SDK's delta call results in an IllegalArgumentException.

**Comment by greg-db** [comment]
No update for this quite yet.

**Comment by krieb** [comment]
@jacobras This should be fixed as of v2.0.4.

**Comment by krieb** [comment]
My mistake, @jacobras, the fix exists in v2.0.4 of the SDK, but the fix to the server-side validation hasn't rolled out to production yet. I will close this issue once the server-side validation fix is live.

**Comment by krieb** [comment]
OK, server side fix is now out. **NOW** it should be fully fixed by updating to **2.0.4**.

**Comment by jacobras** [comment]
Thanks @greg-db and @krieb! I'll update as soon as https://github.com/dropbox/dropbox-sdk-java/issues/57 is fixed.
